rosidl_generator_cpp/resource/action.hpp.em

```diff
@@ -0,0 +1,33 @@
+// generated from rosidl_generator_cpp/resource/action.hpp.em
+// generated code does not contain a copyright notice
+
+@#######################################################################
+@# EmPy template for generating <action>.hpp files
+@#
+@# Context:
+@#  - spec (rosidl_parser.ActionSpecification)
+@#    Parsed specification of the .action file
+@#  - subfolder (string)
+@#    The subfolder / subnamespace of the message, usually 'action'
+@#  - get_header_filename_from_msg_name (function)
+@#######################################################################
+@
+@{
+header_guard_parts = [
+    spec.pkg_name, subfolder,
+    get_header_filename_from_msg_name(spec.action_name) + '_hpp']
+header_guard_variable = '__'.join([x.upper() for x in header_guard_parts]) + '_'
+}@
+#ifndef @(header_guard_variable)
+#define @(header_guard_variable)
+
+#include <action_msgs/msg/goal_info.hpp>
+#include <action_msgs/msg/goal_status_array.hpp>
+#include <action_msgs/srv/cancel_goal.hpp>
+#include <@(spec.pkg_name)/@(subfolder)/@(get_header_filename_from_msg_name(spec.action_name))__goal.hpp>
+#include <@(spec.pkg_name)/@(subfolder)/@(get_header_filename_from_msg_name(spec.action_name))__result.hpp>
+#include <@(spec.pkg_name)/@(subfolder)/@(get_header_filename_from_msg_name(spec.action_name))__feedback.hpp>
+#include <@(spec.pkg_name)/@(subfolder)/@(get_header_filename_from_msg_name(spec.action_name))__struct.hpp>
```

Comment:
Looking at the generated headers for messages and services, it looks like `_struct.hpp` is the only include needed, the rest can be removed.

<details><summary>Generated builtins.hpp which only includes struct and traits</summary>

```C++
// generated from rosidl_generator_cpp/resource/msg.hpp.em
// generated code does not contain a copyright notice

#ifndef TEST_MSGS__MSG__BUILTINS_HPP_
#define TEST_MSGS__MSG__BUILTINS_HPP_

#include "test_msgs/msg/builtins__struct.hpp"
#include "test_msgs/msg/builtins__traits.hpp"

#endif  // TEST_MSGS__MSG__BUILTINS_HPP_
```
</details>